types/report.ts

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { DocumentData } from 'firebase/firestore';
+import 'react-native-get-random-values';
+import { v4 as uuidv4 } from 'uuid';
+import { LazyObject, User, Post, Comment } from '../types';
+
+export class Report extends LazyObject {
+  public reporter?: User;
+  public reported?: User;
+  public content?: Post | Comment | User;
+  public timestamp?: Date;
+
+  public initWithDocumentData(data: DocumentData): void {
+    this.reporter = new User(data.user);
+    this.reported = new User(data.user);
+    if (data.content instanceof Post) this.content = new Post(data.mod);
```

Comment:
Did you test this? no way this works... right? `data.content` is a documentReference. If you tested this and it works, then disregard, otherwise I think this can be handled in one of two ways:

- just have 3 separate optional fields repotedPost, reportedUser, reportedComment
- data.content is a documentReference, so you can get the path in that documentReference and check if the collection is `users/`, `comments/`, or `posts/`

I'm generally reviewing these PRs assuming you've tested them, so i'm hoping that's the case, and this works as intended, but if so, i'm extremely surprised

Reply:
this might be hard to read but this is some peace of mind for you showing i did test. I usually do, i just forgor on the initWithDocumentDatas the other day
![Screenshot (149)](https://user-images.githubusercontent.com/73561858/219894090-0c998d2f-8754-4069-a326-fb25d893915b.png)

Reply:
no prob man <3 yeah that's fuckin impossible to read :p